packages/svelte-scoped/preprocessor/src/index.ts

```diff
@@ -0,0 +1,76 @@
+import type { PreprocessorGroup } from 'svelte/types/compiler/preprocess'
+import { type UnoGenerator, type UserConfig, type UserConfigDefaults, createGenerator, warnOnce } from '@unocss/core'
+import presetUno from '@unocss/preset-uno'
+import { loadConfig } from '@unocss/config'
+import { transformClasses } from './transformClasses'
+import { transformApply } from './transformApply'
+import type { SvelteScopedContext, UnocssSveltePreprocessOptions } from './types'
+
+export * from './types.d.js'
+
+export default function UnocssSveltePreprocess(options: UnocssSveltePreprocessOptions = {}, unoContextFromVite?: SvelteScopedContext): PreprocessorGroup {
+  if (!options.classPrefix)
+    options.classPrefix = 'spu-'
+
+  let uno: UnoGenerator
+
+  return {
+    markup: async ({ content, filename }) => {
+      if (!uno)
+        uno = await getGenerator(options.configOrPath, unoContextFromVite)
+
+      return await transformClasses({ content, filename: filename || '', uno, options })
+    },
+
+    style: async ({ content, attributes }) => {
+      const addPreflights = !!attributes['uno:preflights']
+      const addSafelist = !!attributes['uno:safelist']
+
+      const checkForApply = options.applyVariables !== false
+
+      const changeNeeded = addPreflights || addSafelist || checkForApply
+      if (!changeNeeded)
+        return
+
+      if (!uno)
+        uno = await getGenerator(options.configOrPath)
+
+      let preflightsSafelistCss = ''
+      if (addPreflights || addSafelist) {
+        if (unoContextFromVite)
+          warnOnce('Do not place preflights or safelist within an individual component as they already placed in your global styles injected into the head tag. These options are only for component libraries.')
+        const { css } = await uno.generate([], { preflights: addPreflights, safelist: addSafelist, minify: true })
+        preflightsSafelistCss = css
+      }
+
+      if (checkForApply) {
+        const transformedApplies = await transformApply({
+          code: content,
+          uno,
+          applyVariables: options.applyVariables,
+        })
+
+        if (transformedApplies)
+          return { code: preflightsSafelistCss + transformedApplies }
```

Comment:
just came across this and wanted to mention that svelte preprocessors can (and should) return a sourcemap for the transforms they did. 

https://svelte.dev/docs#compile-time-svelte-preprocess:~:text=Preprocessor%20functions%20should%20additionally%20return%20a%20map%20object%20alongside%20code%20and%20dependencies%2C%20where%20map%20is%20a%20sourcemap%20representing%20the%20transformation.

It might not be needed today as there are more issues with vite css sourcemaps, but if you have the maps ready, it would be great to return them here.

Reply:
Good catch @dominikg! 

Yes, I was under the impression that CSS sourcemaps gave no benefit, but that makes sense to future proof it. In this context is it important to add the filename as the source?

`map: s.generateMap({ hires: true, source: filename || '' }),` in https://github.com/unocss/unocss/pull/2552/commits/dce1a8085c669c7e9d9673683729660f3d434004

Reply:
i think svelte would add it but to be safe, add it. vite-plugin-svelte currently postprocesses the maps to ensure relative paths https://github.com/sveltejs/vite-plugin-svelte/blob/5e9518b3a5e887efbfada483ec0fcdbbc875a35c/packages/vite-plugin-svelte/src/utils/sourcemaps.ts#L41

you can use vite css.devSourceMap option to test it. eg like https://github.com/sveltejs/vite-plugin-svelte/tree/main/packages/e2e-tests/css-dev-sourcemap

Reply:
Ok, I'll leave it in.